docs: Add installation option description (#11735)
argoproj/argo-workflows: docs/installation.md

* A **namespace install** only executes workflows in the namespace it is installed in (typically `argo`). Look for `namespace-install.yaml` in the [release assets](https://github.com/argoproj/argo-workflows/releases/latest).

> * A **managed namespace install**: only executes workflows in a specific namespace ([learn more](managed-namespace.md)).

Reviewer: ```suggestion
* A **managed namespace install**: only executes workflows in a separate namespace from the one it is installed in See [Managed Namespace](managed-namespace.md) for more details.
```

Author: Thank you for your detailed feedback  @agilgur5! I applied your suggestion, so result are as follows.

This is `managed-namespace.md`
<img width="749" alt="image" src="https://github.com/argoproj/argo-workflows/assets/39519869/38d2d509-f06a-4167-83d7-7607151dbdff">

This one is changed description with link of managed-namespace.md in `installation.md`
<img width="698" alt="image" src="https://github.com/argoproj/argo-workflows/assets/39519869/c5deaf4e-4164-4ba6-93b9-accbef1f6963">

Reviewer: Awesome, looks great and thanks for providing the screenshots too!

> !!! Warning "namespace install vs. managed namespace install"

Reviewer: ```suggestion
!!! Info "namespace install vs. managed namespace install"
```

I think [info](https://squidfunk.github.io/mkdocs-material/reference/admonitions/#supported-types) makes more sense here as opposed to a warning. There is nothing to warn about, this is just descriptive

>     A namespace install allows Workflows to run only in the namespace where Argo Workflows is installed. A managed namespace install allows Workflows to run only in one namespace besides the one where Argo Workflows is installed. Using a managed namespace install might make sense if you want some users/processes to be able to run Workflows without granting them any privileges in the namespace where Argo Workflows is installed.  

Reviewer: ```suggestion
    A namespace install allows Workflows to run only in the namespace where Argo Workflows is installed.
    A managed namespace install allows Workflows to run only in one namespace _other than_ the one where Argo Workflows is installed.
    You can use a managed namespace install if you want some users or services to run Workflows without granting them privileges in the namespace where Argo Workflows is installed.  
```

- [Use simpler & more direct language](https://kubernetes.io/docs/contribute/style/style-guide/#use-simple-and-direct-language)
- "processes" -> "services", so as to not confuse folks with Unix processes. "Service" as in "ServiceAccount" fits better in k8s
- tiny formatting changes

Reviewer: The last sentence here can be moved to `managed-namespace.md`. Could put it in the same paragraph as the example that is to be moved.

The first two sentences I think are redundant with `managed-namespace.md`

>     For example, if you only run CI/CD-related Workflows that are maintained by the same team that manages the Argo Workflows installation, it's probably reasonable to use a namespace install. But if all the Workflows are run by a separate data science team, it probably makes sense to give them a data-science-workflows namespace and run a "managed namespace install" of Argo Workflows from another namespace.

Reviewer: ```suggestion
    For example, if you only run CI/CD Workflows that are maintained by the same team that manages the Argo Workflows installation, you may want a namespace install.
    But if all the Workflows are run by a separate data science team, you may want to give them a "data-science-workflows" namespace and use a managed namespace install of Argo Workflows in another namespace.
```
- [Use simpler & more direct language](https://kubernetes.io/docs/contribute/style/style-guide/#use-simple-and-direct-language)
- "managed namespace install" is not quoted anywhere else, so remove quotes there. instead quote "data-science-workflows"
- tiny formatting changes

Reviewer: Let's move this to `managed-namespace.md` instead

>     To configure a managed namespace install, edit the workflow-controller and argo-server Deployments to pass the --managed-namespace argument.

Reviewer: ```suggestion
    To configure a managed namespace install, edit the `workflow-controller` and `argo-server` Deployments to pass the `--managed-namespace` flag.
```

- add backticks around naming
- "argument" -> "flag"

Reviewer: `managed-namespace.md` describes this better with a YAML code block, so I don't this is needed
